src/many-abci/src/module.rs

```diff
@@ -102,6 +119,79 @@ fn _tm_query_from_many_filter(
     Ok(query)
 }
 
+fn transform_list_args(
+    blockchain::ListArgs {
+        count,
+        order,
+        filter,
+    }: blockchain::ListArgs,
+) -> Result<(u64, Order, Query), ManyError> {
+    filter
+        .map_or(
+            Ok(DEFAULT_BLOCK_LIST_QUERY.clone()),
+            _tm_query_from_many_filter,
+        )
+        .map(|filter| {
+            (
+                count.map_or(MAXIMUM_BLOCK_COUNT, |c| {
+                    std::cmp::min(c, MAXIMUM_BLOCK_COUNT)
+                }),
+                order.map_or(tendermint_rpc::Order::Ascending, _tm_order_from_many_order),
+                filter,
+            )
+        })
+}
+
+fn create_pagination(count: u64) -> (u64, u64) {
+    // We can get maximum u8::MAX blocks per page and a maximum of u32::MAX pages
+    // Find the correct number of pages and count
+    let maximum_8_bit_integer: u64 = u8::MAX.into();
+    (
+        num_integer::div_ceil(count, maximum_8_bit_integer),
+        std::cmp::max(count, maximum_8_bit_integer),
```

Comment:
Since the theoretical max is `u8::MAX`, you can return `count` as a `u8` using `as u8`.

Reply:
This is one of the places where deferring the downcast of `count` to `u8` until later simplified the calculations done here.